pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -944,6 +944,14 @@ func (e *BuildExpression) MaybeUpdateTimestamp(ts strfmt.DateTime) error {
 	return nil
 }
 
+func (e *BuildExpression) Copy() (*BuildExpression, error) {
+	bytes, err := json.Marshal(e)
+	if err != nil {
+		return nil, errs.Wrap(err, "Failed to marshal build expression during copy")
+	}
+	return New(bytes)
```

Comment:
Note I believe you can do this more cheaply by dereferencing and creating a new reference, eg.

```
v := *e
return &v
```

Reply:
I'm pretty sure that's just copying the top-level BuildExpression struct, not all the structs that live inside it as pointers.